### PR TITLE
[GH-147] Trying to Fix the deploy workflow action

### PR DESCRIPTION
**Pull request by goulter**
**Change Description:**
This may not have always been the case, but the last few times we've tried to used this workflow, it doesn't work.
I discovered a few things, one being (I think) it was always deploying to dev, no matter what.

**Closes Jira(s)**: 
GH-147

## UW-Directory Pull Request checklist

- [x] I have run `poetry run tox`
- [x] I have selected a `semver-guidance:` label for this pull request (under labels,
      to the right of the screen)

### CODE REVIEW

**Comment by jdiverp** [comment]
I have no concerns but someone closer to software development should do an approve here.

File: scripts/get-deployed-version.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    echo "No target stage provided. Cannot continue."
+    return 1
+  fi
+}
+
+
+function get_stage_version {
+  local stage=$1
+  local url=$(get_stage_url $stage)
+  local version=$(curl -Ssl ${url}/status | grep version | cut -f2 -d: | sed 's| ||g')
```
Comment (miker985) [inline scripts/get-deployed-version.sh:56]: any chance you could include an example "version" line from a status endpoint as a comment?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            # Get the target version from the poetry configuration
+            source ./scripts/globals.sh
+            target_version=$(get_poetry_version)
+          else
+            target_version=$(./scripts/get-deployed-version.sh -s ${source_stage})
+          fi
+          echo "::set-output name=target-version::${target_version}"
+      # For shared instances (dev, eval, prod) we create a slack notification
+      # so that others can be aware of the change.
+      # TODO: Eval and prod deployment notifications should go to #iam-uwnetid
```
Comment (miker985) [inline .github/workflows/deploy.yml:88]: Is there an issue associated with this TODO?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      # This provides at-a-glance links and details in the slack notification for
+      # traceability.
+      # The output winds up reading something like:
+      #   "Deployment workflow for UW Husky Directory app image started by goulter [Audit Record]" where
+      # all of "deployment workflow" "app image" "goulter" and "Audit Record" are all hyperlinks.
+      - if: steps.slack.outputs.canvas-id
+        uses: UWIT-IAM/actions/update-slack-workflow-canvas@0.1
+        env:
+          workflow_link: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+          # image_link: https://gcr.io/uwit-mci-iam/identity-uw.app:${{ steps.reconcile-version.outputs.target-version }}
```
Comment (miker985) [inline .github/workflows/deploy.yml:130]: dead line?

**Comment by miker985** [comment]
All the pieces look fine to me and the workflows are nicely documented.

I'm quite novice at writing github workflows so I'm unable to comment on better ways to do things or possible syntactic issues,

**Review by miker985 — APPROVED** [review]

File: scripts/globals.sh [hunk 4]
```diff
@@ -9,6 +9,17 @@ function get_instance_version {
     echo "$version"
 }
 
+function get_stage_url {
+  case "$1" in
+    dev|eval|prod)
+      echo "https://directory.iam${1}.s.uw.edu"
+      ;;
+    *)
+      echo "invalid stage provided to scripts/globals.sh/get_stage_url"
```
[the remark this answers is not in the document]
Comment (goulter) [reply scripts/globals.sh:18]: This block was inherited, but I can do something better on line 18.

File: scripts/get-deployed-version.sh (hunk 1 above)
Comment (goulter) [reply scripts/get-deployed-version.sh:56]: Thanks Mike, yup, sure can.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (goulter) [reply .github/workflows/deploy.yml:88]: Thanks Mike! There is now ;) https://github.com/UWIT-IAM/uw-husky-directory/issues/152

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (goulter) [reply .github/workflows/deploy.yml:130]: Thanks Mike, yes it is.

**Comment by goulter** [comment]
> All the pieces look fine to me and the workflows are nicely documented.
> 
> I'm quite novice at writing github workflows so I'm unable to comment on better ways to do things or possible syntactic issues,

Cool, then I'm in good company ;) This is a port from another workflow in our identity.uw code base. I understand it somewhat better now. Thanks for the input.

**Review by miker985 — APPROVED** [review]

**Pull request merged**